gubled/gubled_test.go

```diff
@@ -140,7 +140,13 @@ func TestGcmOnlyStartedIfEnabled(t *testing.T) {
 	defer mockCtrl.Finish()
 
 	routerMock := NewMockPubSubSource(mockCtrl)
+
+	amMock := NewMockAccessManager(mockCtrl)
+	msMock := NewMockMessageStore(mockCtrl)
+
 	routerMock.EXPECT().KVStore().Return(store.NewMemoryKVStore(), nil)
+	routerMock.EXPECT().AccessManager().Return(amMock, nil).MaxTimes(2)
+	routerMock.EXPECT().MessageStore().Return(msMock, nil).MaxTimes(2)
```

Comment:
I will add this first.

I have an idea here to extend the mock from the current mock generated in which the call to these methods return the supplied objects.